src/components/Common/NumericInput.js

```diff
@@ -12,78 +12,111 @@ export class NumericInput extends Component {
     super(props)
 
     this.state = {
-      value: '',
-      pristine: true,
-      valid: INVALID
+      value: props.value || '',
+      pristine: props.pristine || true,
+      valid: props.valid || VALID
     }
   }
 
   onPaste = e => {
-    if (isNaN(parseFloat(e.clipboardData.getData('text/plain'))))
+    if (isNaN(parseFloat(e.clipboardData.getData('text/plain')))) {
       e.preventDefault()
+    }
   }
 
   onKeyPress = e => {
-    if (!this.props.acceptFloat) {
-      const firstCharacterIsMinus = e.key === '-' && this.state.value === ''
-      const acceptsNegativeValues = this.props.min < 0 || this.props.max < 0
+    const { min, max, acceptFloat } = this.props
+    const allowsNegative = min < 0 || max < 0
 
-      if (!(firstCharacterIsMinus && acceptsNegativeValues) && isNaN(parseInt(e.key, 10))) e.preventDefault()
+    if (!acceptFloat && !allowsNegative) {
```

Comment:
ohh, good catch!

I did change how this check is done. It will allow `-` symbol to be entered if floats or negative values are allowed. But will prevent any other special char (`e`, `.`, `+`) if float is not allowed.